config/params/config.go

```diff
@@ -201,3 +203,56 @@ func (b *BeaconChainConfig) InitializeForkSchedule() {
 	// Set Bellatrix fork data.
 	b.ForkVersionSchedule[bytesutil.ToBytes4(b.BellatrixForkVersion)] = b.BellatrixForkEpoch
 }
+
+// ForkScheduleEntry is a Version+Epoch tuple for sorted storage in an OrderedForkSchedule
+type ForkScheduleEntry struct {
+	Version [4]byte
+	Epoch   types.Epoch
+}
+
+// OrderedForkSchedule provides a type that can be used to sort the fork schedule and find the Version
+// the chain should be at for a given epoch (via VersionForEpoch).
+type OrderedForkSchedule []ForkScheduleEntry
+
+// Len implements the Len method of sort.Interface
+func (o OrderedForkSchedule) Len() int { return len(o) }
+
+// Swap implements the Swap method of sort.Interface
+func (o OrderedForkSchedule) Swap(i, j int) { o[i], o[j] = o[j], o[i] }
+
+// Less implements the Less method of sort.Interface
+func (o OrderedForkSchedule) Less(i, j int) bool { return o[i].Epoch < o[j].Epoch }
+
+var VersionForEpochNotFound = errors.New("could not find an entry in the ForkVersionSchedule")
+
+// VersionForEpoch finds the Version of the last entry in the last that comes before the given epoch
+func (o OrderedForkSchedule) VersionForEpoch(epoch types.Epoch) ([4]byte, error) {
+	for i := range o {
+		// moving from lowest to highest, only consider epochs that start at or before the requested epoch
+		if epoch < o[i].Epoch {
+			continue
+		}
+		// `i+1 == len(o)`: at last element, don't need to check if there is another fork schedule entry in between
+		// `o[i+1].Epoch > fse.Epoch`: next version's epoch is past the epoch in question, so `i` is correct
+		if i+1 == len(o) || epoch < o[i+1].Epoch {
+			return o[i].Version, nil
+		}
+	}
+	var nope [4]byte
+	return nope, errors.Wrapf(VersionForEpochNotFound, "no epoch in list <= %d", epoch)
+}
+
+// Converts the ForkVersionSchedule map into a list of Version+Epoch values, ordered by Epoch from lowest to highest.
+// See docs for OrderedForkSchedule for more detail on what you can do with this type.
+func (b *BeaconChainConfig) OrderedForkSchedule() OrderedForkSchedule {
+	ofs := make(OrderedForkSchedule, 0)
```

Comment:
The length is known beforehand so you could preallocate it, it probably wont make a difference though

Reply:
Right, I usually prefer appending over pre-allocating and inserting at index for non performance-critical code, because it is harder to implement a bug, and it doesn't make a meaningful difference in practice.

Reply:
Ok to leave as is, IMO